selfservice/flow/login/hook.go

```diff
@@ -171,7 +171,14 @@ func (e *HookExecutor) PostLoginHook(w http.ResponseWriter, r *http.Request, g n
 		return nil
 	}
 
-	if err := e.d.SessionManager().UpsertAndIssueCookie(r.Context(), w, r, s); err != nil {
+	var cookieModifiers []x.CookieModifier
+	// Use nonce cookie modifier when Re-Auth or session upgrade and then issue cookie.
+	// Using nonce modifies the cookie value without affecting the session attributes
+	if a.Refresh || a.RequestedAAL > s.AuthenticatorAssuranceLevel { // TODO: Change to OR to allow block exec
```

Comment:
What about this TODO? What speaks against using a new cookie identifier? Can we just use `x.CookieValuesWithNonce()`?

Reply:
The `TODO` comment wasn't cleaned up after I got it working. Considering - https://github.com/ory/kratos/pull/2761#discussion_r994251413, this `if` seems unnecessary now. I'll remove it.